src/l1_fetcher.rs

```diff
@@ -152,7 +152,9 @@ impl L1Fetcher {
                 .to_block(current_l1_block_number + BLOCK_STEP);
 
             // Grab all relevant logs.
-            let logs = self.get_logs(&filter).await?;
+            let logs = self
+                .retry_call::<Vec<Log>, _>(|| async { provider.get_logs(&filter).await })
```

Comment:
This seems a bit weird, do we really need to specify all that? Also, I don't think we need to await the code inside the closure here already since we're awaiting in the `retry_call` function anyways, so we can probably remove the whole async block.